protobuf/transaction.proto

```diff
@@ -26,68 +26,46 @@ import "protobuf/query.proto";
 
 package grakn.protocol;
 
+service TransactionService {
+
+    // Opens a bi-directional stream representing a stateful transaction, streaming
+    // requests and responses back-and-forth. The first request message must
+    // be {Transaction.Open.Req}. Closing the stream closes the transaction.
+    rpc stream (stream Transaction.Req) returns (stream Transaction.Res);
+}
+
 message Transaction {
 
     message Req {
-        map<string, string> metadata = 1000;
+        string id = 1;
+        map<string, string> metadata = 2;
+        int32 latencyMillis = 3;
```

Comment:
Each request must now be uniquely identified by ID in order for the new Map-based structure to work on the client side. This allows concurrent requests to be processed out-of-order and still handled correctly.

The network latency should be sent with every request that supports iteration. It is used by the server's Iterator to correctly compensate for the network latency.